docs/user-guide/src/verify/assert_assume.md

```diff
@@ -0,0 +1,43 @@
+# Assertions and Assumptions 
+
+> **Note**: Currently, the  [UNSAFE_CORE_PROOF](https://viperproject.github.io/prusti-dev/dev-guide/config/flags.html#unsafe_core_proof)
+> configuration flag must be enabled to use assertions and assumptions.
+
+You can use Prusti to verify that a certain property holds at a certain point
+within the body of a function (via an assertion). It is also possible to
+instruct Prusti to assume that a property holds at a certain point within a
+function (via an assumption).
+
+## Assertions
+
+The `prusti_assert!()` macro instructs Prusti to verify that a certain property
+holds at a specific point within the body of a function. In contrast to the
+`assert!()` macro, which only accepts Rust expressions, `prusti_assert!()`
+accepts [specification](../syntax.md) expressions as arguments. Therefore,
```

Comment:
Although we state in the syntax page that Prusti expressions are side-effect free, we should also say it here. For example `assert!(map.insert(5));` would not work with `prusti_assert!`